src/Platform/Microsoft.Testing.Platform/Logging/FileLoggerProvider.cs

```diff
@@ -29,11 +39,11 @@ public FileLoggerProvider(string logFolder, IClock clock, LogLevel logLevel, str
 
     public LogLevel LogLevel { get; }
 
-    public FileLogger FileLogger { get; private set; }
+    public FileLogger? FileLogger { get; private set; }
 
     public bool SyncFlush { get; }
 
-    public async Task CheckLogFolderAndMoveToTheNewIfNeededAsync(string testResultDirectory)
+    public /* for testing */ virtual async Task CheckLogFolderAndMoveToTheNewIfNeededAsync(string testResultDirectory)
```

Comment:
I am not super comfortable to make something virtual for testing.

Reply:
Same as the comment below, if we don't mark this `virtual` then we cannot mock it and this means the scenarios to test the aggregated configuration will be more complex since we'll have to pass legit paths around and account for the work this method does too.

Reply:
Reverted.